Add support for boot scripts.

Index: u-boot/include/configs/am335x_evm.h
===================================================================
--- u-boot.orig/include/configs/am335x_evm.h
+++ u-boot/include/configs/am335x_evm.h
@@ -113,6 +113,11 @@
 	"loadramdisk=load mmc ${mmcdev} ${rdaddr} ramdisk.gz\0" \
 	"loadimage=load mmc ${bootpart} ${loadaddr} ${bootdir}/${bootfile}\0" \
 	"loadfdt=load mmc ${bootpart} ${fdtaddr} ${bootdir}/${fdtfile}\0" \
+	"script=boot.scr\0" \
+	"loadbootscript=" \
+		"load mmc ${bootpart} ${loadaddr} ${script};\0" \
+	"bootscript=echo Running bootscript from mmc ...; " \
+		"source ${loadaddr}\0" \
 	"mmcloados=run mmcargs; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
 			"if run loadfdt; then " \
@@ -138,6 +143,15 @@
 				"echo Running uenvcmd ...;" \
 				"run uenvcmd;" \
 			"fi;" \
+			"if run loadbootscript; then " \
+				"echo Loaded script from ${script};" \
+				"run bootscript;" \
+			"fi;" \
+			"setenv script /boot/${script};" \
+			"if run loadbootscript; then " \
+				"echo Loaded script from ${script};" \
+				"run bootscript;" \
+			"fi;" \
 			"if run loadimage; then " \
 				"run mmcloados;" \
 			"fi;" \
