Index: u-boot/include/configs/ti_armv7_common.h
===================================================================
--- u-boot.orig/include/configs/ti_armv7_common.h	2013-10-26 08:38:27.433333811 -0700
+++ u-boot/include/configs/ti_armv7_common.h	2013-10-26 08:38:37.000000000 -0700
@@ -149,6 +149,7 @@
 #define CONFIG_CMD_ASKENV
 #define CONFIG_CMD_ECHO
 #define CONFIG_CMD_BOOTZ
+#define CONFIG_SUPPORT_RAW_INITRD
 
 /*
  * Common filesystems support.  When we have removable storage we
