Description: Enable generic tools build
Author: Hector Oron <zumbi@debian.org>

--- u-boot-2014.10~rc2+dfsg1.orig/tools/Makefile
+++ u-boot-2014.10~rc2+dfsg1/tools/Makefile
@@ -52,7 +52,7 @@ HOSTCFLAGS_xway-swap-bytes.o := -pedanti
 hostprogs-y += mkenvimage
 mkenvimage-objs := mkenvimage.o os_support.o lib/crc32.o
 
-hostprogs-y += dumpimage mkimage
+hostprogs-y += dumpimage mkimage mksunxiboot kwboot
 hostprogs-$(CONFIG_FIT_SIGNATURE) += fit_info fit_check_sign
 
 FIT_SIG_OBJS-$(CONFIG_FIT_SIGNATURE) := common/image-sig.o
